include/eve/algo/preprocess_range.hpp

```diff
@@ -7,10 +7,110 @@
 //==================================================================================================
 #pragma once
 
+#include <eve/algo/converting_iterator.hpp>
+#include <eve/algo/detail/preprocess_range.hpp>
```

Comment:
Because I need convert in preprocess_rangr and preprocess_range roughly everywhere.
Like a `relaxed_range` is something that has preprocess range defined for it, that kind of stuff.